phoenix-core/src/main/java/org/apache/phoenix/schema/SchemaExtractionTool.java

```diff
@@ -0,0 +1,473 @@
+            ImmutableBytesWritable key = entry.getKey();
+            ImmutableBytesWritable value = entry.getValue();
+            if(Bytes.toString(key.get()).contains("coprocessor") || Bytes.toString(key.get()).contains(IS_META)) {
+                continue;
+            }
+            defaultProps.put(Bytes.toString(key.get()), "false");
+            definedProps.put(Bytes.toString(key.get()), Bytes.toString(value.get()));
+        }
+    }
+
+    private void setHColumnFamilyProperties(HColumnDescriptor columnDescriptor) {
+        Map<ImmutableBytesWritable, ImmutableBytesWritable> propsMap = columnDescriptor.getValues();
+        for (Map.Entry<ImmutableBytesWritable, ImmutableBytesWritable> entry : propsMap.entrySet()) {
+            ImmutableBytesWritable key = entry.getKey();
+            ImmutableBytesWritable value = entry.getValue();
+            definedProps.put(Bytes.toString(key.get()), Bytes.toString(value.get()));
+        }
+    }
+
+    private void setPTableProperties(PTable table) {
+        Map <String, String> map = table.getValues();
+        for(Map.Entry<String, String> entry : map.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            if(value != null) {
+                definedProps.put(key, value);
+            }
+        }
+    }
+
+    private HTableDescriptor getHTableDescriptor(ConnectionQueryServices cqsi, PTable table)
+            throws SQLException, IOException {
+        return cqsi.getAdmin().getTableDescriptor(
+                TableName.valueOf(table.getPhysicalName().getString()));
+    }
+
+    private String convertPropertiesToString() {
+        StringBuilder optionBuilder = new StringBuilder();
+
+        for(Map.Entry<String, String> entry : definedProps.entrySet()) {
+            String key = entry.getKey();
+            String value = entry.getValue();
+            if(value!=null && defaultProps.get(key) != null && !value.equals(defaultProps.get(key))) {
+                if (optionBuilder.length() != 0) {
+                    optionBuilder.append(", ");
+                }
+                optionBuilder.append(key+"="+value);
+            }
+        }
+        return optionBuilder.toString();
+    }
+
+    private PTable getPTable(String pSchemaName, String pTableName) throws SQLException {
+        String pTableFullName = SchemaUtil.getQualifiedTableName(pSchemaName, pTableName);
+        return getPTable(pTableFullName);
+    }
+
+    private ConnectionQueryServices getCQSIObject() throws SQLException {
+        try(Connection conn = getConnection()) {
+            return conn.unwrap(PhoenixConnection.class).getQueryServices();
+        }
+    }
+
+    public static Connection getConnection() throws SQLException {
+        return ConnectionUtil.getInputConnection(conf);
+    }
+
+    private String getColumnInfoString(PTable table, PTable baseTable) {
```

Comment:
As mentioned above also, it might be cleaner to refactor this API into two different APIs for table and view and move common functionality to a shared function.